### PR TITLE
fix: Inconsistent behaviour with Xcode 16 when `PBXProject.TargetAttributes` is empty 

### PR DESCRIPTION
**Pull request by Speakus**
### Short description 📝

stop generate empty TargetAttributes - because Xcode 16 will always remove such item

### Solution 📦

very simple fix

NB! Better to wait for merging this fix first: https://github.com/fastlane/fastlane/pull/23872

### CODE REVIEW

File: Sources/XcodeProj/Objects/Project/PBXProject.swift [hunk 1]
```diff
@@ -548,7 +548,7 @@ extension PBXProject: PlistSerializable {
                 (value as? PBXObject)?.reference.value ?? value
             }
         }
-        plistAttributes[PBXProject.targetAttributesKey] = plistTargetAttributes
+        plistAttributes[PBXProject.targetAttributesKey] = plistTargetAttributes.isEmpty ? nil : plistTargetAttributes
```
Comment (kwridan) [inline Sources/XcodeProj/Objects/Project/PBXProject.swift:551]: nit: might be useful to add a test to check the key isn't included when there are no attributes

**Review by kwridan — APPROVED** [review]
Thank you for contributing this @Speakus

**Comment by pepicrft** [comment]
I added some tests. Thanks a lot @Speakus for your contribution 🙏🏼

**Comment by pepicrft** [comment]
@all-contributors add @Speakus for code

**Review by pepicrft — APPROVED** [review]

**Comment by pepicrft** [comment]
@all-contributors add @Speakus for code

**Comment by 2sem** [comment]
no TargetAttributes with 4.43.1
4.22.0 generates it

**Pull request merged**